src/components/header/header.jsx

```diff
@@ -141,26 +142,41 @@ class Header extends Component {
 
   connectionConnected = () => {
     this.setState({ account: store.getStore('account') })
+    this.setAddressEnsName();
   };
 
   connectionDisconnected = () => {
     this.setState({ account: store.getStore('account') })
   }
 
+  setAddressEnsName = async () => {
+    const provider = store.getStore('web3context').library.provider
+    const account = store.getStore('account')
+    const { address } = account
+    const network = provider.networkVersion
+    const ens = new ENS({ provider, network })
+    const addressEnsName = await ens.reverse(address).catch(() => {})
+    if (addressEnsName) {
+      this.setState({ addressEnsName })
+    }
+  }
+
   render() {
     const {
       classes
     } = this.props;
 
     const {
       account,
+      addressEnsName,
       modalOpen
     } = this.state
 
     var address = null;
     if (account.address) {
       address = account.address.substring(0,6)+'...'+account.address.substring(account.address.length-4,account.address.length)
     }
+    const addressAlias = addressEnsName || address
```

Comment:
Utilize ENS name if available, otherwise fallback to truncated address (0xadEa...fab9)